src/Engine/Scene/SkeletonComponent.cpp

```diff
@@ -158,8 +167,11 @@ void SkeletonComponent::update( Scalar t ) {
 
     m_animationTime = m_speed * t;
     Scalar lastTime = 0;
-    for ( auto boneAnim : m_animations[m_animationID] ) {
-        lastTime = std::max( lastTime, *boneAnim.getTimes().rbegin() );
+    if ( !m_animations.empty() ) {
+        // m_animationID is always < m_animation.size() unless m_animations.empty()
+        for ( auto boneAnim : m_animations[m_animationID] ) {
```

Comment:
```suggestion
        for ( const auto& boneAnim : m_animations[m_animationID] ) {
```
There are several loops on this file that should benefit from this king a fixes. Use const ref whenever it is the meaning of the loop variable.

Reply:
maybe out of scope of this pr, but ok for this one

Reply:
When encountering some problems not so hard to fix nor so numerous, I prefer to fix them when discovered but it's just a matter of taste.

Reply:
Sure, feel free to do it, it's quicker to fix/commit it directly I think, than comment and ask for modification.
Unless you think we should discuss the proposed modification.
(btw I've done it for this file).